server/the_commander.py

```diff
@@ -0,0 +1,173 @@
+import re
+import shlex
+import subprocess
+
+from sqlalchemy import desc
+
+import king_phisher.plugins as plugin_opts
+import king_phisher.server.database.models as db_models
+import king_phisher.server.plugins as plugins
+import king_phisher.server.signals as signals
+
+
+class Plugin(plugins.ServerPlugin):
+    authors = ['Corey Gilks']
+    title = 'The Commander'
+    description = """
+    Execute an action from the KP Server after new credentials are received. Originally this plugin was created to
+    quickly authenticate to the targets VPN after new credentials are received. When the target is using
+    MFA every second counts, so action must be taken quickly. Operators may be unable to respond fast enough therefore
+    this plugin is needed.
+    
+    You can dynamically include the username, password and MFA values in your command by using the following python 
```

Comment:
Food for thought: if you used Jinja in the template you could adjust the username, password and MFA values. This might be helpful if you got a username submitted that was an email and wanted to trim off the `@domain.com` portion.

If you wanted to go this way, there's a base environment available in [`king_phisher.templates.TemplateEnvironmentBase`](https://king-phisher.readthedocs.io/en/latest/king_phisher/templates.html#king_phisher.templates.TemplateEnvironmentBase).

Reply:
I think I'll stick with the kwargs and keep things light on the configuration side. I figure if someone wants to verify data looks a certain way (domain\username or username@domain.com) they can do some client side validation in the html using jinja.

Which actually means an additional option for stripping domains has to added. The plugin is currently stripping all domains if they exist (L141).